src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -2,20 +2,73 @@
 
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
 
-import static org.junit.Assert.assertTrue;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Currency;
 
-public class ArrayListProductDaoTest
-{
-    private ProductDao productDao;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ArrayListProductDaoTest {
+    @Mock
+    private Product product1;
+    @Mock
+    private Product product2;
+    @Mock
+    private Product product3;
+    @InjectMocks
+    private ArrayListProductDao productDao = new ArrayListProductDao();
 
     @Before
     public void setup() {
-        productDao = new ArrayListProductDao();
+        productDao.getProductList().addAll(Arrays.asList(product1, product2));
+
+        when(product1.getId()).thenReturn(1L);
+        when(product1.getPrice()).thenReturn(new BigDecimal(100));
+        when(product1.getStock()).thenReturn(10);
+
+        when(product2.getId()).thenReturn(2L);
+        when(product2.getPrice()).thenReturn(new BigDecimal(200));
+        when(product1.getStock()).thenReturn(20);
+
+        when(product3.getId()).thenReturn(3L);
     }
 
     @Test
     public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+        assertFalse(productDao.findProducts().isEmpty());
+    }
+
+    @Test
+    public void testNullPrice() {
+        assertFalse(productDao.findProducts().stream()
+                .anyMatch(product -> product.getPrice() == null)
+        );
+    }
+
+    @Test
+    public void testGetProduct() {
+        assertNotNull(productDao.getProduct(1L));
+    }
+
+    @Test
+    public void testSaveProduct() {
+        productDao.save(product3);
+        assertSame(productDao.getProduct(product3.getId()), product3);
```

Comment:
Don't use business logic methods in assertions. Instead, verify that the product list now contains product3